src/hooks/useTransactionConfirmation.ts

```diff
@@ -0,0 +1,37 @@
+import { useState, useCallback, useEffect } from 'react'
+import { ChainId } from '@sushiswap/sdk'
+import { useActiveWeb3React } from './index'
+import useInterval from 'hooks/useInterval'
+
+interface GetTransactionCall {
+  confirmations: number
+  requiredConfirmations: number
+  done: boolean
+}
+
+type ChainConfirmationList = {
+  readonly [chainId in ChainId]?: number
+}
+
+const CHAIN_REQUIRED_CONFIRMATIONS: ChainConfirmationList = {
+  [ChainId.MAINNET]: 20,
+  [ChainId.MATIC]: 192
+}
+
+export default function useTransactionConfirmation(txHash: string): GetTransactionCall {
+  const { library, chainId } = useActiveWeb3React()
+
+  const [confirmations, setConfirmations] = useState(0)
```

Comment:
Should we default to a higher number just in case?
```suggestion
  const [confirmations, setConfirmations] = useState(12)
```